File: Frontend/complaint-system/src/types/general/event.ts
export interface Event {
  id: number;
  event_name: string;
  description: string | null;
  date: Date;
  location: string | null;
  created_at: Date;
  updated_at: Date;
  media: EventMedia[];
}

export interface EventMedia{
  id: number;
  event_id: number;
  media_url: string;
  media_type: string;
  uploaded_at: Date;
}

File: Backend/app/models/event.py
from app.database.database import Base
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime, timezone

class Event(Base):
    __tablename__ = 'events'

    id = Column(Integer, primary_key=True, index=True)
    uploader_id = Column(Integer, ForeignKey('user.id'))
    event_name = Column(String, index=True)
    description = Column(String)
    date = Column(DateTime(timezone=True))
    location = Column(String)
    created_at = Column(DateTime(timezone=True), default=datetime.now(timezone.utc))
    updated_at = Column(DateTime(timezone=True), default=datetime.now(timezone.utc), onupdate=datetime.now(timezone.utc))

    media = relationship("EventMedia", back_populates="event", cascade="all, delete-orphan")
    uploader = relationship("User", back_populates="events", foreign_keys=[uploader_id])